third_party/docfx-doclet-143274/src/main/java/com/microsoft/lookup/BaseLookup.java

```diff
@@ -313,4 +327,110 @@ private String getSeeTagRef(SeeTree seeTree) {
+            .collect(Collectors.toList());
+
+    if (annotationNames.stream().anyMatch("InternalApi"::equals)
+        || annotationNames.stream().anyMatch("InternalExtensionOnly"::equals)) {
+      return "internal";
+    }
+    if (annotationNames.stream().anyMatch("Deprecated"::equals)
+        || hasDeprecatedJavadocTag(element)) {
+      return "deprecated";
+    }
+    if (annotationNames.stream().anyMatch("ObsoleteApi"::equals)) {
+      return "obsolete";
+    }
+    if (annotationNames.stream().anyMatch("BetaApi"::equals)) {
+      return "beta";
+    }
+    return null;
+  }
+
+  public String getStatusComment(T element) {
+    Map<String, Optional<String>> annotationComments = getAnnotationComments(element);
+
+    // Deprecated comments are determined by the Javadoc @deprecated block tag.
+    // See this#replaceBlockTags
+
+    List<String> comments = new ArrayList<>();
+    if (annotationComments.containsKey("InternalApi")) {
+      comments.add(createInternalOnlyNotice(annotationComments.get("InternalApi")));
+    }
+    if (annotationComments.containsKey("InternalExtensionOnly")) {
+      comments.add(
+          createInternalExtensionOnlyNotice(annotationComments.get("InternalExtensionOnly")));
+    }
+    if (annotationComments.containsKey("ObsoleteApi")) {
+      comments.add(createObsoleteNotice(annotationComments.get("ObsoleteApi")));
+    }
+    if (annotationComments.containsKey("BetaApi")) {
+      comments.add(createBetaNotice(annotationComments.get("BetaApi")));
+    }
+
+    if (comments.isEmpty()) {
+      return null;
+    }
+    return String.join("\n\n", comments);
+  }
+
+  private String createBetaNotice(Optional<String> customComment) {
+    return "<aside class=\"beta\">\n"
+        + "<p><strong>Beta</strong></p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "<p>This feature is covered by the <a href=\"/terms/service-terms#1\">Pre-GA Offerings "
+        + "Terms</a> of the Terms of Service. Pre-GA libraries might have limited support, and "
+        + "changes to pre-GA libraries might not be compatible with other pre-GA versions. For "
+        + "more information, see the launch stage descriptions.</p>\n"
+        + "</aside>\n";
+  }
+
+  private String createObsoleteNotice(Optional<String> customComment) {
+    return "<aside class=\"deprecated\">\n"
+        + "<p><strong>Obsolete</strong></p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "<p>This feature is stable for usage in this major version, but may be deprecated in a "
+        + "future release.</p>\n"
+        + "</aside>\n";
+  }
+
+  private String createInternalExtensionOnlyNotice(Optional<String> customComment) {
+    return "<aside class=\"special\">\n"
+        + "<p><strong>Internal Extension Only</strong>: This feature is stable for usage, but is "

```

Comment:
naive question - what does it mean for something to be able to be used, but not implemented? Or more generally; when would `InternalExtensionOnly` be used?

Reply:
InternalExtensionOnly means "You can see this class and use it, but don't create your own class that extends or implements it. This is intended to be extended + implemented only by classes in this same jar. For example, we may add methods to this interface and implement them in our own classes at the same time; but this would be seen as a breaking change to any classes implementing the interface outside of the jar."

See also https://github.com/search?q=org%3Agoogleapis+InternalExtensionOnly&type=code